### ISSUE TITLE
TMSTileSource uses XYZ scheme instead of TMS scheme [BUG]

### ISSUE THREAD

**Issue by njanakiev** [bokeh/bokeh#10102]
Python 3.7.6
Bokeh 2.0.1
HoloViews 1.13.2

I noticed when using `bokeh.models.tiles.TMSTileSource`, that the tiles were misaligned on the y coordinates and were using the XYZ scheme which is also used in the `bokeh.models.tiles.WMTSTileSource` (i.e. the example below had the same output for both `TMSTileSource` and `WMTSTileSource`). Here is a minimal example of that:

```python
import holoviews as hv
import bokeh.models.tiles
hv.extension('bokeh')

URL = 'http://a.tile.osm.org/{Z}/{X}/{Y}.png'
hv.Tiles(bokeh.models.tiles.TMSTileSource(url=URL)).opts(width=600, height=400)
```
This is the output for both the `TMSTileSource` and `WMTSTileSource`:

![Selection_024](https://user-images.githubusercontent.com/9904103/83424367-985d7780-a42c-11ea-8e1e-1c00744b087b.png)

I was unable to find a public data set showing the misalignment. [GeoServer](http://geoserver.org/) can offer TMS and [GDAL2Tiles](https://gdal.org/programs/gdal2tiles.html) creates per default Tiles following the TMS standard. Here is the same example using Leaflet:

```js
var layer = L.tileLayer('http://a.tile.osm.org/{z}/{x}/{y}.png', {
    tms: true, 
    opacity: 0.7, 
    attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors', 
    minZoom: 0, 
    maxZoom: 5
});
```

Here the resulting map for the same URL in Leaflet (as this is not a TMS source):

![Selection_023](https://user-images.githubusercontent.com/9904103/83424044-26852e00-a42c-11ea-8f2e-614e60e04247.png)

Here is the Specification of TMS: https://wiki.osgeo.org/wiki/Tile_Map_Service_Specification

**Comment by bryevdv** [comment]
@brendancol are you able to comment on this?

**Comment by bryevdv** [comment]
Possibly useful ref: https://alastaira.wordpress.com/2011/07/06/converting-tms-tile-coordinates-to-googlebingosm-tile-coordinates/

@njanakiev Is this just a matter of inverting the Y coordinate in your opinion?

**Comment by njanakiev** [comment]
@bryevdv yes, it seems to be only that Y starts from the bottom instead from what I could gather online. I assume this would be the conversion:

```
y = (2^z) - y - 1
```

Here is the diagram from [Tile Map Service Specification](https://wiki.osgeo.org/wiki/Tile_Map_Service_Specification):

![](https://wiki.osgeo.org/images/e/e7/Tms.png)

**Comment by brendancol** [comment]
@mattpap @bryevdv I'm happy to take a look this week and submit PR or review other PR.

**Comment by bryevdv** [comment]
@brendancol  👋 👋 😄 A PR would be fantastic and appreciated, if you have a chance!

**Comment by bryevdv** [comment]
Unfortunately we should have insisted on a pure-Bokeh at the time, something we would do nowadays. Unfortunately I don't know what to do with this issue now to be able to actually judge if there is still a problem, or whether any particular  change might be a successful fix. I am going to close with `NEEDS MORE INFO` we can certainly re-open with a proper reproducer.
